test/test_e2e.rb

```diff
@@ -104,4 +109,27 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert_equal 1, queue.fail_fast
+    assert_operator queue.unprocessed_jobs.length, :<=, 3
+    assert_operator queue.processed_jobs.length, :<=, 2
```

Comment:
Ditto. We should assert this is either `2` or `3` (assuming we set `--max-requeues 1` as suggested above).